awx/api/views/__init__.py

```diff
@@ -1700,47 +1740,79 @@ def _get_insights(self, url, username, password):
                 license
             )
         }
-        return session.get(url, headers=headers, timeout=120)
 
-    def get_insights(self, url, username, password):
+        return headers
+
+    def _get_platform_id(self, host, session, headers):
+        url = '{}/api/inventory/v1/hosts?insights_id={}'.format(
+            settings.INSIGHTS_URL_BASE, host.insights_system_id)
+        res = self._call_insights_api(url, session, headers)
         try:
-            res = self._get_insights(url, username, password)
-        except requests.exceptions.SSLError:
-            return (dict(error=_('SSLError while trying to connect to {}').format(url)), status.HTTP_502_BAD_GATEWAY)
-        except requests.exceptions.Timeout:
-            return (dict(error=_('Request to {} timed out.').format(url)), status.HTTP_504_GATEWAY_TIMEOUT)
-        except requests.exceptions.RequestException as e:
-            return (dict(error=_('Unknown exception {} while trying to GET {}').format(e, url)), status.HTTP_502_BAD_GATEWAY)
+            platform_id = res['results'][0]['id']
+        except (IndexError, KeyError):
+            raise NotFound(_('This host is not recognized as an Insights host.'))
```

Comment:
Let's make this message different than the case where `host.insights_system_id = None`. Maybe `Could not translate insights system id {} into an insights platform id`

Reply:
Sounds good to me.